x-pack/plugins/transform/public/app/sections/transform_management/components/transform_list/expanded_row_preview_pane.tsx

```diff
@@ -4,218 +4,39 @@
-                case ES_FIELD_TYPES.BOOLEAN:
-                  column.dataType = 'boolean';
-                  break;
-                case ES_FIELD_TYPES.DATE:
-                  column.align = 'right';
-                  column.render = (d: any) =>
-                    formatHumanReadableDateTimeSeconds(moment(d).unix() * 1000);
-                  break;
-                case ES_FIELD_TYPES.BYTE:
-                case ES_FIELD_TYPES.DOUBLE:
-                case ES_FIELD_TYPES.FLOAT:
-                case ES_FIELD_TYPES.HALF_FLOAT:
-                case ES_FIELD_TYPES.INTEGER:
-                case ES_FIELD_TYPES.LONG:
-                case ES_FIELD_TYPES.SCALED_FLOAT:
-                case ES_FIELD_TYPES.SHORT:
-                  column.dataType = 'number';
-                  break;
-                case ES_FIELD_TYPES.KEYWORD:
-                case ES_FIELD_TYPES.TEXT:
-                  column.dataType = 'string';
-                  break;
-              }
-            }
-
-            return column;
-          });
-
-          setPreviewData(resp.preview);
-          setColumns(tableColumns);
-          setSortField(sortField);
-          setSortDirection(sortDirection);
-        }
-        concurrentLoads--;
-
-        if (concurrentLoads > 0) {
-          concurrentLoads = 0;
-          getPreview();
-        }
-      } catch (error) {
-        setIsLoading(false);
-        setErrorMessage(
-          i18n.translate('xpack.transform.transformList.stepDetails.previewPane.errorMessage', {
-            defaultMessage: 'Preview could not be loaded',
-          })
-        );
-      }
-    };
-  };
-
-  useRefreshTransformList({ onRefresh: getPreviewFactory() });
-
-  const pagination = {
-    initialPageIndex: pageIndex,
-    initialPageSize: pageSize,
-    totalItemCount: previewData.length,
-    pageSizeOptions: [10, 20],
-    hidePerPageOptions: false,
-  };
-
-  const sorting = {
-    sort: {
-      field: sortField as string,
-      direction: sortDirection,
-    },
-  };
-
-  const onTableChange = ({
-    page = { index: 0, size: 10 },
-    sort = { field: columns[0].field, direction: SORT_DIRECTION.ASC },
-  }: {
-    page?: { index: number; size: number };
-    sort?: { field: keyof typeof previewData[number]; direction: SortDirection | Direction };
-  }) => {
-    const { index, size } = page;
-    setPageIndex(index);
-    setPageSize(size);
-
-    const { field, direction } = sort;
-    setSortField(field);
-    setSortDirection(direction);
-  };
+  const previewConfig = applyTransformConfigToDefineState(
+    getDefaultStepDefineState({} as SearchItems),
+    transformConfig
+  );
 
-  const transformTableLoading = previewData.length === 0 && isLoading === true;
-  const dataTestSubj = `transformPreviewTabContent${!transformTableLoading ? ' loaded' : ''}`;
+  const indexPatternTitle = Array.isArray(transformConfig.source.index)
+    ? transformConfig.source.index.join(',')
+    : transformConfig.source.index;
```

Comment:
potentially might be an issue if the joint index patter does not exist (has a different order etc)

Reply:
yes this can happen. in case there's no match or the source data has been deleted an error like this will be shown:

![image](https://user-images.githubusercontent.com/230104/76314520-40a0eb80-62d7-11ea-8d87-eef7f183f055.png)